mesop/components/sidenav/sidenav.py

```diff
@@ -1,15 +1,45 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Literal
+
 import mesop.components.sidenav.sidenav_pb2 as sidenav_pb
 from mesop.component_helpers import (
   Style,
   insert_composite_component,
+  register_event_handler,
+  register_event_mapper,
   register_native_component,
 )
+from mesop.events import MesopEvent
+
+
+@dataclass(kw_only=True)
+class SidenavOpenedChangedEvent(MesopEvent):
+  """Event representing the opened state change of the sidenav component.
+
+  Attributes:
+      opened: A boolean indicating whether the sidenav component is opened (True) or closed (False).
+      key (str): key of the component that emitted this event.
+  """
+
+  opened: bool
+
+
+register_event_mapper(
+  SidenavOpenedChangedEvent,
+  lambda event, key: SidenavOpenedChangedEvent(
+    key=key.key,
+    opened=event.bool_value,
+  ),
+)
 
 
 @register_native_component
 def sidenav(
   *,
   opened: bool = True,
+  disable_close: bool = False,
```

Comment:
I'm leaning towards we should default this to True because `on_opened_changed` is an optional param (and we can't make it a required param at this point).

As the original issue stated, it's kinda weird to allow the user to Esc and close the sidenav when there isn't proper state handling.

Reply:
Yeah that's a good point. Updated.